miner: fix empty mempool case for waitNext()

Sjors opened the pull request:
Block template fees are calculated by looping over `new_tmpl->vTxFees` and return (early) once the `fee_threshold` is exceeded.

This left an edge case when the mempool is empty, which this commit fixes and adds a test for.

Also update `test/functional/interface_ipc.py` to reflect the new behavior,

Fixes https://github.com/Sjors/sv2-tp/issues/9

Sjors commented:
This is not worth the effort to backport imo.

sedited reviewed (approved):
ACK 2e8fff3f17366e9c2ba054023ad8bd89ac51d584

Sjors committed:
miner: fix empty mempool case for waitNext()

Block template fees are calculated by looping over new_tmpl->vTxFees
and return (early) once the fee_threshold is exceeded.

This left an edge case when the mempool is empty, which this commit
fixes and adds a test for. It does so by using std::accumulate instead
of manual loops.

Also update interface_ipc.py to account for the new behavior.

Co-authored-by: Raimo33 <claudio.raimondi@protonmail.com>

sipa reviewed (commented):
utACK 8f7673257a1a86717c1d83770dc857fc254df107

zaidmstrr reviewed (commented):
Concept ACK [8f76732](https://github.com/bitcoin/bitcoin/pull/33566/commits/8f7673257a1a86717c1d83770dc857fc254df107)

zaidmstrr commented on src/node/miner.cpp:
```diff
@@ -522,18 +523,13 @@ std::unique_ptr<CBlockTemplate> WaitAndCreateNewBlock(ChainstateManager& chainma
-                if (new_fees >= current_fees + options.fee_threshold) return new_tmpl;
-            }
+            // Check if fees increased enough to return the new template
+            const CAmount new_fees = std::accumulate(new_tmpl->vTxFees.begin(), new_tmpl->vTxFees.end(), CAmount{0});
```
Thanks for the benchmarking. But I think to simulate the actual mempool behaviour is complicated. The order of transactions inside mempool is not very deterministic based on fee. So we can't predict that our threshold always will be hit at the last stages of search, even though it searches till the last in most of the cases, but that's not for all.  The function also uses polling, so if you increase the number of executions, the time will increase a bit. But as Sjors said, this calculation will be removed by cluster mempool work later on. So I think we can go with this for now.
